recipes/magnum/all/conanfile.py

```diff
@@ -0,0 +1,710 @@
+            del self.options.target_headless  # Requires EGL (when used with_gl_info)
+            del self.options.with_glxapplication
+            del self.options.with_cglcontext
+            del self.options.with_windowlesscglapplication
+
+        if self.settings.os == "Linux":
+            del self.options.with_cglcontext
+            del self.options.with_windowlesscglapplication
+            del self.options.with_wglcontext
+            del self.options.with_windowlesswglapplication
+            del self.options.with_windowlesswindowseglapplication
+        
+        if self.settings.os == "Macos":
+            del self.options.with_eglcontext
+            del self.options.target_gles
+            del self.options.target_gles2
+            del self.options.with_glxapplication  # Requires GL/glx.h (maybe XQuartz project)
+            del self.options.with_xeglapplication
+            del self.options.with_windowlesseglapplication
+            del self.options.with_windowlessglxapplication  # Requires GL/glx.h (maybe XQuartz project)
+            del self.options.with_windowlesswglapplication
+            del self.options.with_windowlesswindowseglapplication
+            del self.options.target_headless  # Requires EGL (when used with_gl_info)
+
+        if self.settings.os != "Android":
+            del self.options.with_androidapplication
+
+        if self.settings.os != "Emscripten":
+            del self.options.with_emscriptenapplication
+
+        if self.settings.os != "iOS":
+            del self.options.with_windowlessiosapplication
+
+    @property
+    def _executables(self):
+        all_execs = ("gl-info", "al-info", "distancefieldconverter", "fontconverter", "imageconverter", "sceneconverter")
+        return [it for it in all_execs if self.options.get_safe("with_{}".format(it.replace("-", "_")))]
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+    
+    def requirements(self):
+        self.requires("corrade/{}".format(self.version))
+        if self.options.with_gl:
+            self.requires("opengl/system")
+        if self.options.with_vk:
+            self.requires("vulkan-loader/1.2.182")
+
+        if self.options.get_safe("with_eglcontext", False) or \
+           self.options.get_safe("with_xeglapplication", False) or \
+           self.options.get_safe("with_windowlesseglapplication", False) or \
+           self.options.get_safe("with_windowlessiosapplication") or \
+           self.options.get_safe("with_windowlesswindowseglapplication", False) or \
+           self.options.get_safe("target_headless", False):
+            self.requires("egl/system")
+
+        if self.options.with_glfwapplication:
+            self.requires("glfw/3.3.4")
+
+        if self.options.with_sdl2application:
+            self.requires("sdl/2.0.16")
+
+    def build_requirements(self):
+        self.build_requires("corrade/{}".format(self.version))
```

Comment:
Shouldn't this be added conditionally when using 2 profile build?
I had to do this in the `autoconf` recipe:
```suggestion
        if hasattr(self, "settings_build"):
            self.build_requires("corrade/{}".format(self.version))
```

Reply:
There is no reason to do it, specially when the versions of `magnum` and `corrade` (both the requires and the build-requires) will probably be always the same. It is very unlikely that someone will override the `corrade` requirement. I really prefer to keep the recipe simpler.